contracts/multisig_factory.sol

```diff
@@ -1,9 +1,20 @@
 import "basic_multisig.sol";
 
 contract DSMultisigFactory {
+    mapping(address=>bool) is_basic_multisig;
+
     function createBasicMultisig(
         address[] members, uint8 quorum, uint40 window
-    ) returns (DSBasicMultisig) {
-        return new DSBasicMultisig(members, quorum, window);
+    ) returns (DSBasicMultisig ret) {
+        ret = new DSBasicMultisig(members, quorum, window);
+        is_basic_multisig[address(ret)] = true;
+    }
+
+    function isBasicMultisig(address code)
```

Comment:
Better to declare the variable like this:

```
mapping (address => bool) public isBasicMultisig;
```

Then you can omit the function as you will get it for free.